plugins/modules/fabric_virtual_networks_workflow_manager.py

```diff
@@ -0,0 +1,2950 @@
+                    msg = (
+                        "Given parameter '{0}' for the given anycast doesnot match so gateway needs update."
+                    ).format(param)
+                    self.log(msg, "INFO")
+                    return True
+
+        return False
+
+    def get_anycast_gateway_update_payload(self, anycast, anycast_details_in_ccc):
+        """
+        Constructs the payload necessary to update the Anycast Gateway configuration in Cisco Catalyst Center.
+        Args:
+            self (object): An instance of a class used for interacting with Cisco Catalyst Center.
+            anycast (dict): A dictionary containing the new configuration parameters for the Anycast Gateway.
+            anycast_details_in_ccc (dict): A dictionary containing the current Anycast Gateway configuration details from CCC.
+        Returns:
+            dict: A dictionary containing the payload needed to update the Anycast Gateway configuration.
+        Description:
+            This function constructs an update payload for the Anycast Gateway by combining existing values from
+            `anycast_details_in_ccc` with any new values specified in the `anycast` dictionary. If a parameter exists in
+            `anycast`, it is prioritized over the existing value. If a parameter does not exist in `anycast`, the function
+            retains the current value from `anycast_details_in_ccc`.
+            The resulting payload can be used for updating the Anycast Gateway configuration in the Cisco Catalyst Center.
+        """
+
+        vn_name = anycast_details_in_ccc.get("virtualNetworkName")
+        anycast_payload = {
+            "id": anycast_details_in_ccc.get("id"),
+            "fabricId": anycast_details_in_ccc.get("fabricId"),
+            "virtualNetworkName": vn_name,
+            "ipPoolName": anycast_details_in_ccc.get("ipPoolName"),
+            "vlanName": anycast_details_in_ccc.get("vlanName"),
+            "vlanId": anycast_details_in_ccc.get("vlanId"),
+            "isCriticalPool": anycast_details_in_ccc.get("isCriticalPool"),
+            "poolType": anycast_details_in_ccc.get("poolType"),
+            "isIntraSubnetRoutingEnabled": anycast_details_in_ccc.get("isIntraSubnetRoutingEnabled")
+        }
+        params_in_playbook = ["tcp_mss_adjustment", "traffic_type", "security_group_name", "layer2_flooding_enabled",
+                              "fabric_enabled_wireless", "ip_directed_broadcast", "multiple_ip_to_mac_addresses",
+                              "supplicant_based_extended_node_onboarding", "group_policy_enforcement_enabled"]
+
+        anycast_mapping = self.get_anycast_gateway_mapping(vn_name)
+
+        if vn_name == "INFRA_VN":
+            params_to_remove = ["security_group_name", "layer2_flooding_enabled", "fabric_enabled_wireless",
+                                "ip_directed_broadcast", "multiple_ip_to_mac_addresses"]
+            for param in params_to_remove:
```

Comment:
for param in params_to_remove:
            if param in params_in_playbook:
                params_in_playbook.remove(param)
                self.log("Removing parameter '{0}' from payload construction for INFRA_VN.".format(param), "DEBUG")